src/grpc/grpc_server.cc

```diff
@@ -1236,6 +1236,21 @@ CommonHandler::RegisterTrace()
     int32_t count;
     uint32_t log_frequency;
     std::string filepath;
+
+    if (!request.model_name().empty()) {
+      TRITONSERVER_Message* config_message = nullptr;
+      err = TRITONSERVER_ServerModelConfig(
+          tritonserver_.get(), request.model_name().c_str(),
+          -1 /* model version */, 1 /* config_version */, &config_message);
```

Comment:
Since we don't actually need the model config here, I think [TRITONSERVER_ServerModelIsReady](https://github.com/triton-inference-server/core/blob/e456d2ad22778776dfbea631c5d262c42567b6f1/src/tritonserver.cc#L2553) would be a more lightweight proxy for model existing.

Reply:
Interestingly, when I switch to using this function, no error is returned from the core components: 
`bool ready = false;`
`auto err = TRITONSERVER_ServerModelIsReady(server_.get(), model_name.c_str(), -1 /* model version */, &ready);`
The only indication that something went wrong is that the bool parameter `ready` remains false when the model does not exist / flips to true if a model does exist. However, I don't believe this is the desired effect considering the [HandleModelReady](https://github.com/triton-inference-server/server/blob/328614e1bd85997ff8559b934a33eaa636f46b1b/src/http_server.cc#L1486-L1487) function attempts to pass on an error status from `TRITONSERVER_ServerModelIsReady`. I am thinking that this may be an issue from an unhandled exception [here](https://github.com/triton-inference-server/core/blob/e456d2ad22778776dfbea631c5d262c42567b6f1/src/server.cc#L461C2-L470C26) in core. Seems like an error getting the model will have no effect and the function will still return a `Status::Success`?

Reply:
I don't know the original context for why this doesn't return an error if server ready but model not ready. However, you can just check the `ready` arg instead after calling.

As for the model ready handler, it checks [both](https://github.com/triton-inference-server/server/blob/328614e1bd85997ff8559b934a33eaa636f46b1b/src/http_server.cc#L1490-L1491): `(ready && (err == nullptr))`

Reply:
Yes, I can pivot on the ready arg, but it seems strange that this code trace would encounter an error that essentially becomes overwritten, especially when `HandleModelReady` anticipates the possibility of an error. I guess keeping the ready flag false becomes the manifestation of the error...